### PR TITLE
fixing integration tests

### LINKED ISSUE #10: (Seeming non-deterministic) Travis-CI Build Failures
Looking at https://travis-ci.org/nmklotas/GitLabApiClient/pull_requests

I thought my change in https://github.com/nmklotas/GitLabApiClient/pull/9 would address this, but was wrong. Even though it had built successfully multiple times, it again started to fail with 401 and other errors that lead me to believe the GitLab service was not setup properly for the test run.

Will continue discussion here.

### LINKED ISSUE #71: Target Core 3.0?
Consider dropping .NET 4.5.2

Let's target .NET Standard 2.0 and .NET Core 3.0.
Hopefully with .NET Core 3.0 we can slowly convert the API to have [nullable references](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references)

Switch to use [System.Text.Json](https://www.nuget.org/packages/System.Text.Json) for improved performance.

I noticed a 2 second reduction in a simple data set of 1050 slack users with only the slack id and emails attributes while deserializing into a POCO

### PR DESCRIPTION
**Pull request by casz**
So after much trial and error (3 branches)
https://github.com/casz/GitLabApiClient/tree/fix/dockerTests
https://github.com/casz/GitLabApiClient/tree/docker
https://github.com/casz/GitLabApiClient/tree/dockerv2

I finally went GTFO on travis and switched to GitHub action for Linux CI as it behaves as what I could produce locally, so definitely a lot of tests that are not passing.

There are still some issues running Windows CI with containers that is apparent on both AppVeyor and GitHub actions.

@nmklotas This is a step forward at least to fixing the integration tests!

Basically travis seems to have issue collecting xunit's standard output and hosting the gitlab container.

fixes #10 
fixes #44
fixes #71

### CODE REVIEW

**Comment by nmklotas** [comment]
nice, travis does not work good with this

**Comment by casz** [comment]
@nmklotas it became Christmas after all! 70.01% code coverage 😱

**Comment by nmklotas** [comment]
70 cov is a lot :dagger:

**Comment by casz** [comment]
I'd like to rebase and merge to keep commit history.
